src/daemon/rpmostreed-transaction-types.c

```diff
@@ -1694,7 +1697,161 @@ rpmostreed_transaction_new_deploy (GDBusMethodInvocation *invocation,
   return (RpmostreedTransaction *) g_steal_pointer (&self);
 }
 
-/* ================================ InitramfsState ================================ */
+/* ================================ InitramfsEtc ================================ */
+
+typedef struct {
+  RpmostreedTransaction parent;
+  char *osname;
+  char **track;
+  char **untrack;
+  gboolean untrack_all;
+  gboolean force_sync;
+  GVariantDict *options;
+} InitramfsEtcTransaction;
+
+typedef RpmostreedTransactionClass InitramfsEtcTransactionClass;
+
+GType initramfs_etc_transaction_get_type (void);
+
+G_DEFINE_TYPE (InitramfsEtcTransaction,
+               initramfs_etc_transaction,
+               RPMOSTREED_TYPE_TRANSACTION)
+
+static void
+initramfs_etc_transaction_finalize (GObject *object)
+{
+  InitramfsEtcTransaction *self;
+
+  self = (InitramfsEtcTransaction *) object;
+  g_free (self->osname);
+  g_strfreev (self->track);
+  g_strfreev (self->untrack);
+  g_clear_pointer (&self->options, g_variant_dict_unref);
+
+  G_OBJECT_CLASS (initramfs_etc_transaction_parent_class)->finalize (object);
+}
+
+static gboolean
+initramfs_etc_transaction_execute (RpmostreedTransaction *transaction,
```

Comment:
I thought we'd been trending a bit towards glomming everything into `Deploy` so it could be done in one go?  I guess it doesn't much matter for this case, just curious if you had a reason to make this separate.  (It is clearer to read separately, but more duplicated code)